### PR TITLE
[lldb] Simplify check for lldb(-rpc-server)

### PR DESCRIPTION
**Pull request by JDevlieghere**

### CODE REVIEW

**Review by bulbazord — APPROVED** [review]
🚢

**Comment by JDevlieghere** [comment]
@swift-ci please test

File: lldb/source/Commands/CommandObjectHealthcheck.cpp [hunk 1]
```diff
@@ -60,8 +60,7 @@ bool CommandObjectHealthcheck::DoExecute(Args &args,
   // When in an interactive graphical session and not, for example,
   // running LLDB running over ssh, open the log file straight away in
   // the user's configured editor or the default Console.app otherwise.
-  if ((strcmp("lldb", getprogname()) == 0 ||
-       strcmp("lldb-rpc-server", getprogname()) == 0) &&
+  if (llvm::StringRef(getprogname()).starts_with("lldb") &&
```
Comment (adrian-prantl) [inline lldb/source/Commands/CommandObjectHealthcheck.cpp:63]: I'm having a hard time remembering why we were checking `argv[0]` here in the first place.

Comment (adrian-prantl) [reply lldb/source/Commands/CommandObjectHealthcheck.cpp:63]: Ah. The test suite, where it is `python`!

**Pull request merged**